src/main/java/net/dv8tion/jda/internal/entities/EntityBuilder.java

```diff
@@ -1387,11 +1427,13 @@ public ReceivedMessage createMessage(DataObject jsonObject, @Nullable MessageCha
         final long authorId = author.getLong("id");
         MemberImpl member = null;
 
-        if (channel == null && jsonObject.isNull("guild_id") && authorId != getJDA().getSelfUser().getIdLong())
+        if (channel == null && jsonObject.isNull("guild_id"))
         {
             DataObject channelData = DataObject.empty()
-                    .put("id", channelId)
-                    .put("recipient", author);
+                    .put("id", channelId);
+            if (authorId != getJDA().getSelfUser().getIdLong())
+                    channelData.put("recipient", author);
+            //this will construct a minimal private channel if we are the author
             channel = createPrivateChannel(channelData);
```

Comment:
This comment should be retooled to differently explain that while this creates a private channel, it may only be a partial channel if we are the author as we don't have the recipient info.